core/src/main/scala/chisel3/Disable.scala

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import chisel3.internal._
+import chisel3.experimental.{IntrinsicModule, OpaqueType, SourceInfo}
+import chisel3.internal.sourceinfo.SourceInfoTransform
+
+import scala.language.experimental.macros
+import scala.collection.immutable.ListMap
+
+/** API for handling disabling of simulation constructs
+  *
+  * Disables may be non-synthesizable so they can only be used for disabling simulation constructs
```

Comment:
They are fair questions and use cases. At the moment, you can't do anything with it except pass it as an argument (implicitly or explicitly) to the new `chisel3.ltl.*Property` functions, eg. `AssertProperty`.

It's not so much that users necessarily should not be able to use the stuff more flexibly, but more that we don't currently have any other way of stopping them from using it to influence synthesizable hardware which would be bad (you may get a warning/error in synthesis but I'm not sure, but we'd like to protect users from accidentally simulating with it as well). Another possibility would be enforcing this property as a firtool check, but I think it would turn into a global analysis which we'd probably like to avoid if possible.

FWIW, power user library writers like yourself can instantiate the intrinsic directly and then you have the value exposed as a Bool, but I'm trying very hard to keep that aspect of it out of the standard user API, at least for now.